File: public/code/names/1.rb
require "net/http"
require "json"

url = URI.parse("https://api.buttondown.email/v1/subscribers/{SUBSCRIBER_ID}")
http = Net::HTTP.new(url.host, url.port)
http.use_ssl = true
headers = {
  "Authorization" => "Token {API_KEY}",
  "Content-Type" => "application/json"
}
data = {"metadata" => {"first_name" => "Pikachu"}}
request = Net::HTTP::Patch.new(url.request_uri, headers)
request.body = data.to_json
response = http.request(request)
puts(response.code)
puts(response.body)

File: public/code/names/1.py
import requests
import json

url = "https://api.buttondown.email/v1/subscribers/{SUBSCRIBER_ID}"
headers = {"Authorization": "Token {API_KEY}"}
data = {"metadata": {"first_name": "Pikachu"}}
response = requests.patch(url, headers=headers, data=json.dumps(data))
print(response.status_code)
print(response.json())
